modules/pam_shells/pam_shells.c

```diff
@@ -13,27 +13,51 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdbool.h>
 #include <sys/stat.h>
 #include <syslog.h>
 #include <unistd.h>
+#if defined (USE_ECONF)	&& defined (VENDORDIR)
+#include <libeconf.h>
+#endif
 
 #include <security/pam_modules.h>
 #include <security/pam_modutil.h>
 #include <security/pam_ext.h>
 
 #define SHELL_FILE "/etc/shells"
-
+#define SHELLS "shells"
+#define ETCDIR "/etc"
 #define DEFAULT_SHELL "/bin/sh"
 
+static pam_handle_t *log_pamh = NULL;
+
+static bool check_file(const char *filename)
```

Comment:
Stefan, would it be possible to let the callback function have an extra opaque argument that could be specified to the function that calls this callback?
If yes, this would help to avoid adding a static variable to the module - something we strive not  to do nowadays.

Reply:
Yes, that's a good idea. I have implemented it.